### PR TITLE
Adding new experimental resources

### PR DESCRIPTION
**Pull request by nfx**

### CODE REVIEW

File: catalog/resource_external_location.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	return a.client.Post(a.context, "/unity-catalog/external-locations", el, &el)
+}
+
+func (a ExternalLocationsAPI) get(id string) (el ExternalLocationInfo, err error) {
+	err = a.client.Get(a.context, "/unity-catalog/external-locations/"+id, nil, &el)
+	return
+}
+
+func (a ExternalLocationsAPI) update(name string, el ExternalLocationInfo) error {
+	return a.client.Patch(a.context, "/unity-catalog/external-locations/"+name, el)
```
Comment (alexott) [inline catalog/resource_external_location.go:38]: URLencode name? and is it really name in the API? :confused:

File: catalog/resource_external_location.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+func (a ExternalLocationsAPI) delete(name string) error {
+	return a.client.Delete(a.context, "/unity-catalog/external-locations/"+name, nil)
```
Comment (alexott) [inline catalog/resource_external_location.go:42]: same here

File: catalog/resource_external_location.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	Owner          string `json:"owner,omitempty" tf:"computed"`
+	MetastoreID    string `json:"metastore_id,omitempty" tf:"computed"`
+}
+
+func (a ExternalLocationsAPI) create(el *ExternalLocationInfo) error {
+	return a.client.Post(a.context, "/unity-catalog/external-locations", el, &el)
+}
+
+func (a ExternalLocationsAPI) get(id string) (el ExternalLocationInfo, err error) {
+	err = a.client.Get(a.context, "/unity-catalog/external-locations/"+id, nil, &el)
```
Comment (alexott) [inline catalog/resource_external_location.go:33]: here as well, if id == name

File: catalog/resource_external_location.go (hunk 1 above)
Comment (nfx) [reply catalog/resource_external_location.go:38]: yes, it's name. alphanumeric validation may be a better option than urlencoding. i don't think we allow emojis in names here...

Comment (alexott) [reply catalog/resource_external_location.go:38]: External locations support any SQL identifier and any character is allowed if you put them into back quotes ([doc](https://docs.databricks.com/sql/language-manual/sql-ref-identifiers.html#parameters))

**Review by alexott — COMMENTED** [review]
Lgtm, can’t check privileges mapping right now

**Review by alexott — APPROVED** [review]
Lgtm

**Pull request merged**